test/NuGet.Core.Tests/NuGet.Packaging.Test/DependenciesGroupsForEachTFMRuleTests.cs

```diff
@@ -0,0 +1,446 @@
+            var (compat, file, nuspec) = rule.Categorize(fileStrings, frameworks);
+            var issues = rule.GenerateWarnings(compat, file, nuspec).ToList();
+
+            // Assert
+            Assert.False(issues.Any(p => p.Code == NuGetLogCode.NU5128));
+            Assert.False(issues.Any(p => p.Code == NuGetLogCode.NU5130));
+        }
+
+        [Fact]
+        public void CatagorizeTFMs_PackageWithPerfectlyMatchedTFMsInLibAndNuspec_ShouldAllBeEmpty()
+        {
+            //Arrange
+            var frameworks = new NuGetFramework[]
+            {
+                FrameworkConstants.CommonFrameworks.Net2,
+                FrameworkConstants.CommonFrameworks.Net35,
+                FrameworkConstants.CommonFrameworks.Net4,
+                FrameworkConstants.CommonFrameworks.Net45,
+                FrameworkConstants.CommonFrameworks.NetStandard10,
+                FrameworkConstants.CommonFrameworks.NetStandard13,
+                FrameworkConstants.CommonFrameworks.NetStandard20,
+            };
+
+            var fileStrings = new string[]
+            {
+                "lib/net20/test.dll",
+                "lib/net35/test.dll",
+                "lib/net40/test.dll",
+                "lib/net45/test.dll",
+                "lib/netstandard1.0/test.dll",
+                "lib/netstandard1.3/test.dll",
+                "lib/netstandard2.0/test.dll",
+            };
+
+            // Act
+            var rule = new DependenciesGroupsForEachTFMRule();
+            var (compat, file, nuspec) = rule.Categorize(fileStrings, frameworks);
+
+            Assert.Empty(compat);
+            Assert.Empty(file);
+            Assert.Empty(nuspec);
+        }
+
+        [Fact]
+        public void CatagorizeTFMs_PackageWithUnmatchedTFMsInNuspec_ShouldHaveOneNotEmpty()
+        {
+            //Arrange
+            var frameworks = new NuGetFramework[]
+            {
+                FrameworkConstants.CommonFrameworks.Net2,
+                FrameworkConstants.CommonFrameworks.Net35,
+                FrameworkConstants.CommonFrameworks.Net4,
+                FrameworkConstants.CommonFrameworks.Net45,
+                FrameworkConstants.CommonFrameworks.NetStandard10,
+                FrameworkConstants.CommonFrameworks.NetStandard13,
+                FrameworkConstants.CommonFrameworks.NetStandard20,
+            };
+
+            var fileStrings = new string[]
+            {
+                "lib/net20/test.dll",
+                "lib/net35/test.dll",
+                "lib/net40/test.dll",
+                "lib/netstandard1.0/test.dll",
+                "lib/netstandard1.3/test.dll",
+                "lib/netstandard2.0/test.dll",
+            };
+
+            // Act
+            var rule = new DependenciesGroupsForEachTFMRule();
+            var (compat, file, nuspec) = rule.Categorize(fileStrings, frameworks);
+
+            Assert.Empty(compat);
+            Assert.Empty(file);
+            Assert.NotEmpty(nuspec);
```

Comment:
You should assert for the expected values. A bug that causes the wrong TFMs to be put in the result will cause the test to pass.